disco/disco.go

```diff
@@ -109,6 +110,10 @@ func (d *Disco) CredentialsForHost(hostname svchost.Hostname) (auth.HostCredenti
 	if d.credsSrc == nil {
 		return nil, nil
 	}
+	if aliasedHost, aliasExists := d.aliasMap[hostname]; aliasExists {
+		log.Printf("[DEBUG] CredentialsForHost found alias %s for %s", hostname, aliasedHost)
+		hostname = aliasedHost
+	}
```

Comment:
This does seem reasonable to me but if you didn't already I'd suggest checking in the Terraform CLI and TFE provider code to make sure nothing is trying to look up credentials as `d.CredentialsSource().ForHost(hostname)` directly, instead of going indirectly through this wrapper method.

If you do find anything doing that then I'd suggest to change it to use `d.CredentialsForHost` rather than complicating the design here. It seems reasonable to require going through the Disco object to get the alias behavior.

It might be worth also updating the doc comment for `CredentialsSource` above to mention that accessing the credentials source directly means that hostname aliases will be ignored.

Reply:
@apparentlymart I believe that the only code that does this kind of credentials lookup is provider installation through network mirror. That client does not go through a service discovery process and [service discovery is not required in the mirror protocol](https://developer.hashicorp.com/terraform/internals/provider-network-mirror-protocol#protocol-base-url).

However, one might reasonably expect localterraform.com to work within provider_installation { network_mirror block. [Our generic hostname docs](https://developer.hashicorp.com/terraform/cloud-docs/registry/using#generic-hostname-terraform-enterprise) only mention module sources as a target for the generic hostname, which gives some validation to the original approach of only overriding certain services using ForceHostServices. On the other hand, I think provider sources would be a natural use case as well, even if it didn't work as network mirror. What are your thoughts?

Reply:
Now that I think of it, Terraform Enterprise would never be a network mirror, right? So no need to make it work there.

Reply:
Hmm yes the use of our host-based auth scheme for network mirrors is a bit unusual because users specify that as exact URL rather than hostname and there intentionally isn't any service discovery indirection there because the network mirror is only a physical location and not also part of an object's unique identifier as is the case for provider registry addresses.

I agree it seems weird to support localterraform.com in that context, since provider mirror addresses will never appear inside a Terraform module and so there's never any need to share exactly the same address between modules running in different Terraform Enterprise instances.

Seems reasonable to ignore the hostname aliases for auth in that case.